Common/Tools/TrackTuner.h

```diff
@@ -476,38 +493,72 @@ struct TrackTuner : o2::framework::ConfigurableGroup {
-    std::string grDcaZPullNameData = "pullsUpgrDcaZ";
-
-    grDcaZResVsPtPionMC.reset(dynamic_cast<TGraphErrors*>(td->Get(grDcaZResNameMC.c_str())));
-    grDcaZResVsPtPionData.reset(dynamic_cast<TGraphErrors*>(td->Get(grDcaZResNameData.c_str())));
-    grDcaZMeanVsPtPionMC.reset(dynamic_cast<TGraphErrors*>(td->Get(grDcaZMeanNameMC.c_str())));
-    grDcaZMeanVsPtPionData.reset(dynamic_cast<TGraphErrors*>(td->Get(grDcaZMeanNameData.c_str())));
-    grDcaZPullVsPtPionMC.reset(dynamic_cast<TGraphErrors*>(td->Get(grDcaZPullNameMC.c_str())));
-    grDcaZPullVsPtPionData.reset(dynamic_cast<TGraphErrors*>(td->Get(grDcaZPullNameData.c_str())));
-    if (!grDcaZResVsPtPionMC.get() || !grDcaZResVsPtPionData.get() || !grDcaZMeanVsPtPionMC.get() || !grDcaZMeanVsPtPionData.get() || !grDcaZPullVsPtPionMC.get() || !grDcaZPullVsPtPionData.get()) {
-      LOG(fatal) << "Something wrong with the names of the correction graphs for dcaZ. Fix it!";
+      grDcaXYResVsPtPionMC[iphibin].reset(loadGraph(iphibin, "resCurrentDcaXY").release());
+      grDcaXYResVsPtPionData[iphibin].reset(loadGraph(iphibin, "resUpgrDcaXY").release());
+      grDcaXYMeanVsPtPionMC[iphibin].reset(loadGraph(iphibin, "meanCurrentDcaXY").release());
+      grDcaXYMeanVsPtPionData[iphibin].reset(loadGraph(iphibin, "meanUpgrDcaXY").release());
+      grDcaXYPullVsPtPionMC[iphibin].reset(loadGraph(iphibin, "pullsCurrentDcaXY").release());
+      grDcaXYPullVsPtPionData[iphibin].reset(loadGraph(iphibin, "pullsUpgrDcaXY").release());
+
+      if (!grDcaXYResVsPtPionMC[iphibin].get() || !grDcaXYResVsPtPionData[iphibin].get() || !grDcaXYMeanVsPtPionMC[iphibin].get() || !grDcaXYMeanVsPtPionData[iphibin].get() || !grDcaXYPullVsPtPionMC[iphibin].get() || !grDcaXYPullVsPtPionData[iphibin].get()) {
+        LOG(fatal) << "[TrackTuner]     Something wrong with the names of the correction graphs for dcaXY. Fix it!";
+      }
+    }
+
+    if (inputNphiBins != 0) {
+      LOG(info) << "[TrackTuner]    Loading phi-dependent Z TGraphErrors";
+    }
+    for (int iphibin = 0; iphibin < nPhiBins; ++iphibin) {
+      grDcaZResVsPtPionMC[iphibin].reset(loadGraph(iphibin, "resCurrentDcaZ").release());
+      grDcaZMeanVsPtPionMC[iphibin].reset(loadGraph(iphibin, "meanCurrentDcaZ").release());
+      grDcaZPullVsPtPionMC[iphibin].reset(loadGraph(iphibin, "pullsCurrentDcaZ").release());
+      grDcaZResVsPtPionData[iphibin].reset(loadGraph(iphibin, "resUpgrDcaZ").release());
+      grDcaZMeanVsPtPionData[iphibin].reset(loadGraph(iphibin, "meanUpgrDcaZ").release());
+      grDcaZPullVsPtPionData[iphibin].reset(loadGraph(iphibin, "pullsUpgrDcaZ").release());
+
+      if (!grDcaZResVsPtPionMC[iphibin].get() || !grDcaZResVsPtPionData[iphibin].get() || !grDcaZMeanVsPtPionMC[iphibin].get() || !grDcaZMeanVsPtPionData[iphibin].get() || !grDcaZPullVsPtPionMC[iphibin].get() || !grDcaZPullVsPtPionData[iphibin].get()) {
+        LOG(fatal) << "Something wrong with the names of the correction graphs for dcaZ. Fix it!";
```

Comment:
as above, I'd add here in the message of the fatal a reference to the iPhiBin that causes the issue

```suggestion
        LOG(fatal) << "[TrackTuner] Something wrong with the names of the correction graphs for dcaZ. Fix it!";
```